hedgedoc: fix editor crashing

SuperSandro2000 opened the pull request:
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

winterqt commented:
So did upstream do away with esbuild? Seems like it since you didn't have to add our esbuild.

SuperSandro2000 commented:
No, we are just using the one in node_modules. It does not make sense to vendor it anyway because it must match the lock file.

winterqt commented:
> No, we are just using the one in node_modules. It does not make sense to vendor it anyway because it must match the lock file.

Wonder why we didn't use the one in node_modules all this time then.

drupol commented on pkgs/servers/web-apps/hedgedoc/default.nix:
```diff
@@ -76,10 +61,8 @@ mkYarnPackage rec {
+    # TODO: debug why this file is not included in the upstream webpack bundle.
+    rm node_modules/js-yaml/dist/js-yaml.mjs
 
     npm run build
```
How about using yarn in here ?

drupol replied:
```suggestion
    yarn build
```

winterqt commented:
I changed the title of this PR to remove the leading space. Amazingly, GitHub doesn't trim the PR title upon creation...? @SuperSandro2000 Did you use the website to make this PR? Normally your browser should strip it, so I'm guessing this was made through the API?

drupol commented:
Thanks for taking care of that @SuperSandro2000 !

I tried your branch locally and it works. I also tried to replace `npm run build` with `yarn build` and works as well.

Don't you think we should take the opportunity to update the build script and use yarn for building?

Check the [official installation procedure](https://docs.hedgedoc.org/setup/manual-setup/), I think we should follow those steps.

drupol commented:
I updated my hedgedoc server with the changes from here, and it works. However, when going to the features page, there is an message on the top now.

![Screenshot_20220724-094603_Firefox~2.png](https://user-images.githubusercontent.com/252042/180637564-fe36facc-0be5-4ed4-8cd0-018694126948.png)

I guess this is related to the file we removed.

SuperSandro2000 commented:
> Did you use the website to make this PR?

Probably something off in Refinded GitHub.



> I tried your branch locally and it works. I also tried to replace `npm run build` with `yarn build` and works as well.

I tried that, too and it didn't out of the box. yarn is a bit more tricky to handle and run offline so I would ignore it for now. As mentioned earlier this just runs the build script and the program that is used shouldn't matter.

SuperSandro2000 committed:
hedgedoc: refactor to fix editor crashing, replace inactive maintainer with myself

SuperSandro2000 committed:
yarn2nix: change yarnFlags to append by default

The main usecase for this variable is to append --production to yarn
which now got a whole lot easier because you no longer need to repeat
the defaults.

SuperSandro2000 commented on pkgs/development/tools/yarn2nix-moretea/yarn2nix/default.nix:
```diff
@@ -69,7 +69,7 @@ in rec {
     yarnNix ? mkYarnNix { inherit yarnLock; },
     offlineCache ? importOfflineCache yarnNix,
-    yarnFlags ? defaultYarnFlags,
+    yarnFlags ? [ ],
```
This is not much used in nixpkgs and I am pretty confident that this is a none breaking change. I am still going to remove the extra flags accumulated in other packages after this is merged.

thatsmydoing replied:
We use this in our internal project to remove the `--ignore-scripts` flag when we need to build native extensions for a module. As it is now, it doesn't seem possible to unset any of the default flags so our use case is broken (what's more it's been backported to nixos-22.05).

SuperSandro2000 replied:
Not sure how we build those in nixpkgs. How about adding a boolean variable wether to run scripts?

SuperSandro2000 replied:
#198950

drupol reviewed (approved):
I couldn't agree more with this PR.

We should also fix broken HedgeDoc 1.9.4 so I think we should merge it asap.

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 182643` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
4 packages built:
</summary>
<ul>
<li>hedgedoc</li>
<li>uivonim</li>
<li>vieb</li>
<li>yarn2nix</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
hedgedoc:
<details>

When evaluating attribute ‘hedgedoc’:
warning: EvalError
Cannot evaluate attribute ‘hedgedoc’ in ‘/nix/store/7n5k8glkfw19zn14g0apnrl507sm155l-nixpkgs’.
</details>
uivonim:
<details>

When evaluating attribute ‘uivonim’:
warning: EvalError
Cannot evaluate attribute ‘uivonim’ in ‘/nix/store/7n5k8glkfw19zn14g0apnrl507sm155l-nixpkgs’.
</details>
vieb:
<details>

When evaluating attribute ‘vieb’:
warning: EvalError
Cannot evaluate attribute ‘vieb’ in ‘/nix/store/7n5k8glkfw19zn14g0apnrl507sm155l-nixpkgs’.
</details>
yarn2nix:
<details>

When evaluating attribute ‘yarn2nix’:
warning: EvalError
Cannot evaluate attribute ‘yarn2nix’ in ‘/nix/store/7n5k8glkfw19zn14g0apnrl507sm155l-nixpkgs’.
</details>
</details>

drupol commented:
Excellent, thanks !!!
